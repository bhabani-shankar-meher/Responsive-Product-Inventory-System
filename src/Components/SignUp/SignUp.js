import { Formik, Form, Field } from "formik";
import { Button, CircularProgress } from "@mui/material";
import { useNavigate } from "react-router-dom";
import Box from "@mui/material/Box";
import Grid from "@mui/material/Grid";
import { TextField } from "formik-mui";
import * as Yup from "yup";
import Typography from "@mui/material/Typography";
import Divider from "@mui/material/Divider";

const SignUp = () => {
  const navigate = useNavigate();
  const loginSchema = Yup.object().shape({
    emailID: Yup.string().email().required("Email is Required"),
    password: Yup.string()
      .required("Password is Required")
      .min(8, "Please Enter 8 char")
      .max(16, "Please use under 16"),
    againPassword: Yup.string()
      .required("Password is Required")
      .min(8, "Please Enter 8 char")
      .max(16, "Please use under 16")
      .oneOf([Yup.ref("password"), null], "Passwords must match"),
    firstName: Yup.string().required("First name is Required"),
    lastName: Yup.string().required("Last is Required"),
    location: Yup.string().required("Location is Required"),
    mobile: Yup.string().required("Mobile No. is Required"),
  });

  return (
    <Formik
      initialValues={{
        emailID: "",
        password: "",
        againPassword: "",
        firstName: "",
        lastName: "",
        location: "",
        mobile: "",
      }}
      validationSchema={loginSchema}
      onSubmit={(values, { setSubmitting }) => {
        try {
          values.id = values.emailID;
          if (values) {
            fetch(`http://localhost:5000/login`, {
              method: "POST",
              headers: {
                "Content-Type": "application/json",
              },
              body: JSON.stringify(values, null, 2),
            });
            setSubmitting(false);
            navigate("/login", { replace: true });
          }
        } catch (exception) {
          console.log(exception);
        }
      }}
    >
      {({ submitForm, isSubmitting }) => (
        <Form>
          <Box sx={{ width: "80%", marginLeft: "10%" }} mt={5} mb={3}>
            <Grid
              container
              rowSpacing={3}
              columnSpacing={{ xs: 1, sm: 2, md: 3 }}
            >
              <Grid item xs={12} md={12}>
                <Typography textAlign="center" variant="h4">
                  SignUp
                </Typography>
                <Divider sx={{ width: "100%", marginTop: "20px" }} />
              </Grid>
              <Grid item xs={12} md={12}>
                <Field
                  component={TextField}
                  name="emailID"
                  type="email"
                  label="Email"
                  style={{ width: "100%" }}
                />
              </Grid>
              <Grid item xs={12} md={6}>
                <Field
                  component={TextField}
                  name="password"
                  type="password"
                  label="Password"
                  style={{ width: "100%" }}
                />
              </Grid>
              <Grid item xs={12} md={6}>
                <Field
                  component={TextField}
                  name="againPassword"
                  type="password"
                  label="Re-Password"
                  style={{ width: "100%" }}
                />
              </Grid>
              <Grid item xs={12} md={6}>
                <Field
                  component={TextField}
                  name="firstName"
                  type="text"
                  label="First Name"
                  style={{ width: "100%" }}
                />
              </Grid>
              <Grid item xs={12} md={6}>
                <Field
                  component={TextField}
                  name="lastName"
                  type="text"
                  label="Last Name"
                  style={{ width: "100%" }}
                />
              </Grid>
              <Grid item xs={12} md={6}>
                <Field
                  component={TextField}
                  name="location"
                  type="text"
                  label="Location"
                  style={{ width: "100%" }}
                />
              </Grid>
              <Grid item xs={12} md={6}>
                <Field
                  component={TextField}
                  name="mobile"
                  type="tel"
                  label="Mobile"
                  style={{ width: "100%" }}
                />
              </Grid>
            </Grid>
          </Box>
          <Grid item xs={12} md={12}>
            {isSubmitting && <CircularProgress />}
          </Grid>
          <Box sx={{ width: "80%", marginLeft: "10%" }}>
            <Grid item xs={12} md={12}>
              <Button
                variant="contained"
                color="primary"
                disabled={isSubmitting}
                onClick={submitForm}
              >
                SignUp
              </Button>
              <Button
                variant="outlined"
                color="secondary"
                style={{ marginLeft: "5px" }}
                onClick={() => navigate("/", { replace: true })}
              >
                Back
              </Button>
            </Grid>
          </Box>
        </Form>
      )}
    </Formik>
  );
};

export default SignUp;
