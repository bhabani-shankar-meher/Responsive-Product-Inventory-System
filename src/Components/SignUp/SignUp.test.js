import SignUp from "./SignUp";
import { render, screen } from "@testing-library/react";
import { create } from "react-test-renderer";

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useNavigate: () => jest.fn(),
}));

describe("SIGNUP PAGE SNAPSHOT", () => {
    test("SignUp Page Snapshot Executed Successfully", () => {
        const signUpSnapshot = create(<SignUp />);
        expect(signUpSnapshot).toMatchSnapshot();
    });
});
describe("SIGNUP PAGE TEST", () => {
    test("SignUp Page Heading Test", () => {
        render(<SignUp />);
        expect(
            screen.getByRole("heading", {
                name: /signup/i,
            })
        ).toBeInTheDocument();
    });
    test("Email", () => {
        render(<SignUp />);
        const label = screen.getAllByText(/email/i);
        expect(label).not.toHaveLength(0);
    });
    test("Password", () => {
        render(<SignUp />);
        const label = screen.getAllByText(/password/i);
        expect(label).not.toHaveLength(0);
    });
    test("Re-Password", () => {
        render(<SignUp />);
        const label = screen.getAllByText(/re-password/i);
        expect(label).not.toHaveLength(0);
    });
    test("First Name", () => {
        render(<SignUp />);
        const label = screen.getAllByText(/first name/i);
        expect(label).not.toHaveLength(0);
    });
    test("Last Name", () => {
        render(<SignUp />);
        const label = screen.getAllByText(/last name/i);
        expect(label).not.toHaveLength(0);
    });
    test("Location", () => {
        render(<SignUp />);
        const label = screen.getAllByText(/location/i);
        expect(label).not.toHaveLength(0);
    });
    test("Mobile", () => {
        render(<SignUp />);
        const label = screen.getAllByText(/mobile/i);
        expect(label).not.toHaveLength(0);
    });
    test("Login Button Test", () => {
        render(<SignUp />);
        expect(
            screen.getByRole("button", {
                name: /signup/i,
            })
        ).toBeInTheDocument();
    });
    test("Back Button Test", () => {
        render(<SignUp />);
        expect(
            screen.getByRole("button", {
                name: /back/i,
            })
        ).toBeInTheDocument();
    });
});
