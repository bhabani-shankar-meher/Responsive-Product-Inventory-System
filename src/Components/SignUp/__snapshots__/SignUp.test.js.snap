// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SIGNUP PAGE SNAPSHOT SignUp Page Snapshot Executed Successfully 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="MuiBox-root css-1mo2yih"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-qwfiov-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 css-1idn90j-MuiGrid-root"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 css-149xwtl-MuiTypography-root"
        >
          SignUp
        </h4>
        <hr
          className="MuiDivider-root MuiDivider-fullWidth css-1dh14ht-MuiDivider-root"
        />
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 css-1idn90j-MuiGrid-root"
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor=":r0:"
            id=":r0:-label"
          >
            Email
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id=":r0:"
              name="emailID"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="email"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Email
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor=":r1:"
            id=":r1:-label"
          >
            Password
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id=":r1:"
              name="password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="password"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Password
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor=":r2:"
            id=":r2:-label"
          >
            Re-Password
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id=":r2:"
              name="againPassword"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="password"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Re-Password
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor=":r3:"
            id=":r3:-label"
          >
            First Name
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id=":r3:"
              name="firstName"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  First Name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor=":r4:"
            id=":r4:-label"
          >
            Last Name
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id=":r4:"
              name="lastName"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Last Name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor=":r5:"
            id=":r5:-label"
          >
            Location
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id=":r5:"
              name="location"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Location
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor=":r6:"
            id=":r6:-label"
          >
            Mobile
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id=":r6:"
              name="mobile"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="tel"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Mobile
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 css-13i4rnv-MuiGrid-root"
  />
  <div
    className="MuiBox-root css-lnse6"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 css-13i4rnv-MuiGrid-root"
    >
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        SignUp
      </button>
      <button
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-wnpvqa-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "marginLeft": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        Back
      </button>
    </div>
  </div>
</form>
`;
